axelor-contract/src/main/java/com/axelor/apps/contract/service/ContractServiceImpl.java

```diff
@@ -583,6 +583,8 @@ public InvoiceLine generate(Invoice invoice, ContractLine line) throws AxelorExc
       this.copyAnalyticMoveLines(line.getAnalyticMoveLineList(), invoiceLine);
     }
 
+    invoiceLine.setPriceDiscounted(
+        invoice.getInAti() ? invoiceLine.getInTaxPrice() : invoiceLine.getPrice());
```

Comment:
Pass the value in constructor 
https://github.com/axelor/axelor-open-suite/pull/5432/commits/d032b0018ae7bfb0e1dfc35fb50a501bc42db19c#diff-62529bae7a3a9407ffbb20654177f778R535